driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala

```diff
@@ -0,0 +1,110 @@
+import org.apache.curator.framework.api.ExistsBuilder
+import org.apache.curator.test.TestingServer
+import org.apache.zookeeper.CreateMode
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FlatSpec, _}
+
+/**
+ * This test specifies the behaviour of CuratorFactoryHolder that encapsulates the real curator's factory.
+ * @author anistal
+ */
+@RunWith(classOf[JUnitRunner])
+class CuratorFactoryHolderSpec extends FlatSpec with Matchers with BeforeAndAfter with GivenWhenThen with SLF4JLogging {
+
+  var zkTestServer: TestingServer = _
+
+  before {
+    zkTestServer = new TestingServer(CuratorFactoryHolderSpec.TestServerZKPort)
+    zkTestServer.start()
+
+    val instance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.basicConfig, Some(true))
+    Option(instance.get.checkExists().forPath("/test")) match {
+      case eb: ExistsBuilder =>
+        instance.get.delete().deletingChildrenIfNeeded().forPath(CuratorFactoryHolderSpec.PathTestNode)
+      case None =>
+        log.debug("Test node not created. It is not necessary to delete it.")
+    }
+    CuratorFactoryHolder.resetInstance()
+  }
+
+  after {
+    CuratorFactoryHolder.resetInstance()
+    zkTestServer.stop()
+  }
+
+  "The curator ZK client" must "throws the correct error" in {
+    Given("an empty configuration with a not running ZK server in default port")
+    When("it tries to create the instance and the server is getting down")
+    val instance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.emptyConfig, Some(true))
+    Then("throws a CuratorConnectionLossException")
+    an[CuratorConnectionLossException] should be thrownBy (instance.get.getData.forPath(
+      CuratorFactoryHolderSpec.PathTestNode
+    ))
+  }
+
+  "The curator ZK client" must "create correctly" in {
+    Given(s"ZK configuration: $CuratorFactoryHolderSpec.configString")
+    val instance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.basicConfig)
+    When("creates a ephimeral node in ZK server")
+    instance.get.create().withMode(CreateMode.EPHEMERAL).forPath(CuratorFactoryHolderSpec.PathTestNode)
+    assert(Option(instance.get.checkExists().forPath(CuratorFactoryHolderSpec.PathTestNode)) != None)
+  }
+
+  "The curator ZK client" must "reuse  the same connecton" in {
+    Given(s"ZK configuration: $CuratorFactoryHolderSpec.configString")
+    When("an instance is created with the CuratorFactoryHolder")
+    val instance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.basicConfig)
+    And("other instance is created with the CuratorFactoryHolder")
+    val secondInstance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.basicConfig)
+    Then("the factory return the same instance for both cases")
+    instance should be theSameInstanceAs secondInstance
+  }
+
+  "The curator ZK client" must "not reuse  the same connecton when resetInstance is invoked" in {
```

Comment:
You have to specify the name only the first time. The next test cases should/can start with "it"